state/pgstatestorage/l1infotree.go

```diff
@@ -65,6 +65,11 @@ func (p *PostgresStorage) GetLatestL1InfoRoot(ctx context.Context, maxBlockNumbe
 		return entry, err
 	}
 
+	// L1InfoTreeIndex = 0 is a special case (empty tree) therefore we will set GER as zero
+	if entry.L1InfoTreeIndex == 0 {
+		entry.GlobalExitRoot.GlobalExitRoot = state.ZeroHash
+	}
+
```

Comment:
This business logic make more sense to put in state layer instead of storage.

Reply:
I moved it into the finalizer.